### PR TITLE
Add CDN infra

### PR DESCRIPTION
**Pull request by pamelafox**
This PR adds CDN modules to core/, with a cdn-profile.bicep module, cdn-endpoint.bicep module, and a cdn.bicep module that creates a profile with a *single* endpoint. If a developer wants multiple endpoints, they'll need to use the submodules instead.

Currently, the main customization is the originUrl and deliveryRules, which are the only aspects I've customized in my uses of the CDN modules.

You can see these modules in use here:
https://github.com/pamelafox/flask-charts-api-container-app/blob/868adefb9c612537d0f960f206c426dfbd4c4906/infra/main.bicep#L57

### CODE REVIEW

File: templates/common/infra/bicep/core/cdn/cdn-endpoint.bicep [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+param location string = resourceGroup().location
+param tags object = {}
+
+@description('The origin URL for the endpoint')
+@minLength(1)
+param originUrl string
+
+@description('The name of the CDN profile resource')
+@minLength(1)
+param cdnProfileName string
```
Comment (jongio) [inline templates/common/infra/bicep/core/cdn/cdn-endpoint.bicep:11]: Is this something we can default?

File: templates/common/infra/bicep/core/cdn/cdn-endpoint.bicep [hunk 2]
```diff
@@ -0,0 +1,49 @@
+param name string
```
Comment (jongio) [inline templates/common/infra/bicep/core/cdn/cdn-endpoint.bicep:1]: I'd like to put this in a folder named "networking" instead of cdn.

Comment (pamelafox) [reply templates/common/infra/bicep/core/cdn/cdn-endpoint.bicep]: Sure, moved to networking

File: templates/common/infra/bicep/core/cdn/cdn-endpoint.bicep (hunk 1 above)
Comment (pamelafox) [reply templates/common/infra/bicep/core/cdn/cdn-endpoint.bicep]: Hm, we could default it to empty string, but then the existing resource definition wouldn't work out at the end. I looked at other examples of modules which parent: a resource from another module, and the names don't have default values. Maybe I'm mis-thinking this though, let me know.

File: templates/common/infra/bicep/core/networking/cdn-endpoint.bicep [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      rules: deliveryPolicyRules
+    }
+  }
+}
+
+resource cdnProfile 'Microsoft.Cdn/profiles@2022-05-01-preview' existing = {
+  name: cdnProfileName
+}
+
+output uri string = 'https://${endpoint.properties.hostName}'
```
Comment (jongio) [inline templates/common/infra/bicep/core/networking/cdn-endpoint.bicep:49]: also output id and name

File: templates/common/infra/bicep/core/networking/cdn-profile.bicep [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+resource profile 'Microsoft.Cdn/profiles@2022-05-01-preview' = {
+  name: name
+  location: location
+  tags: tags
+  sku: {
+    name: sku
+  }
+}
+
+output name string  = profile.name
```
Comment (jongio) [inline templates/common/infra/bicep/core/networking/cdn-profile.bicep:32]: output id too

File: templates/common/infra/bicep/core/networking/cdn.bicep [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    name: cdnEndpointName
+    location: location
+    tags: tags
+    cdnProfileName: cdnProfile.outputs.name
+    originUrl: originUrl
+    deliveryPolicyRules: deliveryPolicyRules
+  }
+}
+
+output uri string = cdnEndpoint.outputs.uri
```
Comment (jongio) [inline templates/common/infra/bicep/core/networking/cdn.bicep:34]: output the id / name for each included resource

File: templates/common/infra/bicep/core/cdn/cdn-endpoint.bicep (hunk 1 above)
Comment (jongio) [reply templates/common/infra/bicep/core/cdn/cdn-endpoint.bicep]: Ok

File: templates/common/infra/bicep/core/networking/cdn.bicep (hunk 5 above)
Comment (pamelafox) [reply templates/common/infra/bicep/core/networking/cdn.bicep:34]: Sure, added and did test deploy.

File: templates/common/infra/bicep/core/networking/cdn-endpoint.bicep (hunk 3 above)
Comment (pamelafox) [reply templates/common/infra/bicep/core/networking/cdn-endpoint.bicep:49]: Added and did test deploy.

**Comment by pamelafox** [comment]
Hm, sorry, I didn't intend to remove the other reviewers from the list, Github seemed to remove them when I re-requested a review from @jongio .

File: templates/common/infra/bicep/core/networking/cdn.bicep [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+output profileName string = cdnProfile.outputs.name
```
Comment (jongio) [inline templates/common/infra/bicep/core/networking/cdn.bicep:35]: Please sort all outputs alpha

File: templates/common/infra/bicep/core/networking/cdn.bicep [hunk 7]
```diff
@@ -0,0 +1,38 @@
+// Module to create a CDN profile with a single endpoint
+param cdnProfileName string
+param cdnEndpointName string
+param location string = resourceGroup().location
+param tags object = {}
+
+@description('Origin URL for the CDN endpoint')
+param originUrl string
+
+@description('Delivery policy rules')
```
Comment (jongio) [inline templates/common/infra/bicep/core/networking/cdn.bicep:10]: Please sort alpha

Comment (pamelafox) [reply templates/common/infra/bicep/core/networking/cdn.bicep:10]: To clarify, should all params be alphabetized, or just ones with descriptions?
(I'm looking at other modules but I think they don't follow the style yet)

```
param cdnProfileName string
param cdnEndpointName string
param location string = resourceGroup().location
param tags object = {}

@description('Delivery policy rules')
param deliveryPolicyRules array = []

@description('Origin URL for the CDN endpoint')
param originUrl string
```

Comment (pamelafox) [reply templates/common/infra/bicep/core/networking/cdn.bicep:11]: Okay, I believe everything is alphabetized per preference. I've re-deployed a template with the files with no issue.

**Review by jongio — APPROVED** [review]

**Comment by jongio** [comment]
Thanks!

**Pull request merged**